specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2025-03-01-preview/Issues.json

```diff
@@ -0,0 +1,1248 @@
+          },
+          {
+            "name": "InProgress",
+            "value": "InProgress",
+            "description": "The investigation execution is in progress"
+          },
+          {
+            "name": "Succeeded",
+            "value": "Succeeded",
+            "description": "The investigation execution completed successfully"
+          },
+          {
+            "name": "Failed",
+            "value": "Failed",
+            "description": "The investigation execution failed"
+          },
+          {
+            "name": "Canceled",
+            "value": "Canceled",
+            "description": "The investigation execution was canceled"
+          }
+        ]
+      },
+      "readOnly": true
+    },
+    "InvestigationMetadata": {
+      "type": "object",
+      "description": "Properties of the current investigation",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "uuid",
+          "description": "The unique identifier of the investigation"
+        },
+        "execution": {
+          "$ref": "#/definitions/InvestigationExecution",
+          "description": "The execution details of the investigation"
+        },
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The creation time of the investigation"
+        },
+        "runParameters": {
+          "$ref": "#/definitions/RunParameters",
+          "description": "The parameters that were used to start the investigation"
+        }
+      },
+      "required": [
+        "id",
+        "execution",
+        "createdAt",
+        "runParameters"
+      ]
+    },
+    "InvestigationResult": {
+      "type": "object",
+      "description": "Details about the investigation result",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The identifier of the investigation"
+        },
+        "origin": {
+          "$ref": "#/definitions/Origin",
+          "description": "The origin of the investigation"
+        },
+        "createdAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The creation time of the investigation"
+        },
+        "lastModifiedAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The last update time of the investigation"
+        },
+        "execution": {
+          "$ref": "#/definitions/InvestigationExecution",
+          "description": "The execution details of the investigation"
+        },
+        "runParameters": {
+          "$ref": "#/definitions/RunParameters",
+          "description": "The parameters used to run the investigation"
+        },
+        "children": {
+          "type": "array",
+          "description": "The children of this investigation, findings and evidences",
+          "items": {
+            "$ref": "#/definitions/object"
```

Comment:
>object [](http://example.com/codeflow?start=35&length=6)

why is the schematized object InvestigationEntity not being used ? 

object seems to be a free flow object. all items of children array should be of same schema and having a schema less object can allow for each item to be of different schema, which can cause confusion to customers.

Reply:
The deep structure of the investigation entity is complex and will be prone to many changes, extensions, and enhancements during the preview. So - in the preview API - we decided to avoid officially documenting it so that customers do not create dependencies on its specific structure. Learnings from the preview will help us decide how to properly document this structure towards the GA.